### ISSUE TITLE
Multiple LSP servers attached to the same buffer

### ISSUE THREAD

**Issue by sidlatau** [akinsho/flutter-tools.nvim#171]
Issue is related to PR https://github.com/akinsho/flutter-tools.nvim/pull/163
and  revert of it:
https://github.com/akinsho/flutter-tools.nvim/commit/0bd239706f3a512b4619118a0f1e736976ff4f46

After revert I am getting the previous problem, that multiple LSP servers attach to the same buffer:
<img width="1152" alt="image" src="https://user-images.githubusercontent.com/4973296/181612658-9e0532d7-53b7-4c15-bb3d-7d9736ba5586.png">
Looks like the original fix caused other problems, so I investigated more my config and found that issue disappears if I remove 
https://github.com/dart-lang/dart-vim-plugin from my config.

@akinsho, I see you mentioned this plugin in commit  https://github.com/akinsho/flutter-tools.nvim/commit/0417d4c45e732b5c96f4a42dbb562e5be0af81c1  - is this an intent to move some of `dart-vim-plugin` functionality to `flutter-tools.nvim`? What I miss if I remove `dart-vim-plugin`, is commentstring setup in https://github.com/dart-lang/dart-vim-plugin/blob/master/ftplugin/dart.vim
Or should both plugins work nicely together?

**Comment by akinsho** [comment]
@sidlatau I used to use that plugin as well for the syntax highlighting since hover docs still require old school vim regex highlights, I don't think I noticed a changed with comment strings though, but I'm writing dart quite infrequently now so could just not have hit that edge case. 

I basically copied the relevant synatx highlighting from that plugin into this one because I noticed the same issue that you do with that plugin. I think what it does is that it triggers the ft detection twice using some viml magic that I can't decipher. If you are on nightly, I think this won't be an issue because they now detect race conditions in setting up lsps. If you think the comment string stuff is necessary, I don't mind that logic also being ported into this plugin, so users of this plugin don't need to use both that one and this one.

Added benefit is that it also avoids whatever logic they are running that triggers multiple lsps
